components/Visit/VisitInfoForm.tsx

```diff
@@ -88,9 +91,31 @@ export default function VisitInfoForm({
         if (e.target.type === 'checkbox') {
             value = e.target.checked;
         } else value = e.target.value;
-        const data = transformData({ ...visitData, [key]: value });
-        onChange && onChange(data);
-        setVisitData({ ...visitData, [key]: value });
+
+        if (key === 'householdItem') {
+            const booleanValue = Boolean(value);
+
+            setVisitData((visitData) => ({
+                ...visitData,
+                // reset household items notes if householdItem is unchecked
+                household: !booleanValue ? '' : visitData.household,
+                householdItem: booleanValue,
+            }));
+        } else {
+            setVisitData((visitData) => ({
+                ...visitData,
+                [key]: value,
+            }));
+        }
+
+        if (onChange) {
+            const updatedData = {
+                ...visitData,
+                [key]: value,
+                ...(key === 'householdItem' && !value && { household: '' }),
+            };
+            onChange(transformData(updatedData));
+        }
```

Comment:
could we move this block to into  `transformData` function?
We can have a simple one line "check and set" value similar to line 79:
```ts
kidsQ: visitData.clothingBoy || visitData.clothingGirl ? toInt(visitData.kidsQ) : 0
```

can we do something like:
```ts
household: visitData.houseHoldItem ? visitData.household : ''
```

Reply:
Yes, that'd be more clean and clear! 
I also added initialVisitData?.householdItem for a condition in useState for the case that the checkbox is checked but note is empty.